MeTTaIL/src/main/scala/io/f1r3fly/mettail/InstInterpreterCases.scala

```diff
@@ -721,17 +711,22 @@ object InstInterpreterCases {
                   moduleProcessor: ModuleProcessor
                 ): BasePres = {
     val (modulePath, theoryDecl) =
-      moduleProcessor.resolveDottedPath(resolvedModules, currentModulePath, ctor.dottedpath_).right.get
+      moduleProcessor.resolveDottedPath(resolvedModules, currentModulePath, ctor.dottedpath_) match {
+        case Right(result) => result
+        case Left(msg)     => sys.error(s"Failed to resolve dotted path: $msg")
```

Comment:
This logic should be moved to `checkCtor`. The `handleCtor` function should not fail.